### DISCUSSION TITLE
Mqtt external broker on custom port and topic

### DISCUSSION THREAD

**Discussion by G14nLu** [jblance/mpp-solar#64] [Q&A]
I'm having some trouble to set up external broker and custom port and topic. The documentation is not very complete in this description. Can you help me?

**Comment by jblance** [comment]
Do you to know how to setup the broker or to point to an external broker?
Can you supply more details on your current setup and what command you are using

Thanks

**Reply by G14nLu** [reply]
i want to point an external broker on a specific port, setting my username and password. The command should be something like this: "mpp-solar -n 'FirstInverter' -P PI30 -o screen -c QPIGS -p /dev/ttyUSB0 -q DOMAIN:PORT --mqttuser USERNAME --mqttpass PASSWORD.
Can you help me? 
Many thanks

**Reply by jblance** [reply]
Youll need to use a mqtt output eg -o mqtt also -q is for the broker address, --mqttport is to specify a port
Add -D to get a lot of extra debug info

**Reply by G14nLu** [reply]
`mpp-solar: error: unrecognized arguments: --mqttport #####`

**Reply by G14nLu** [reply]
also another strange error:
`mpp-solar: error: unrecognized arguments: -t basetopic/subtopic`
from the command with `-t "basetopic/subtopic"`

**Reply by jblance** [reply]
what version are you using
can you include the full command and results, so i can see what you are doing 
there is no -t option for mpp-solar command

**Reply by G14nLu** [reply]
> what version are you using
> can you include the full command and results, so i can see what you are doing
> there is no -t option for mpp-solar command

`Solar Device Command Utility, version: 0.7.18, recent changes: refactor mqtt based outputs`

**Reply by jblance** [reply]
try the latest version - there was a bug fix for port recently
